src/Tools/AI Test Toolkit/src/TestSuite/AITTestMethodLines.Page.al

```diff
@@ -0,0 +1,266 @@
+                }
+                field("No. of Tests Failed - Base"; Rec."No. of Tests - Base" - Rec."No. of Tests Passed - Base")
+                {
+                    Editable = false;
+                    ApplicationArea = All;
+                    Caption = 'No. of Tests Failed - Base';
+                    ToolTip = 'Specifies the number of tests that failed in the base Version.';
+                    Style = Unfavorable;
+                    Visible = false;
+
+                    trigger OnDrillDown()
+                    var
+                        AITTestSuite: Record "AIT Test Suite";
+                        AITLogEntry: Codeunit "AIT Log Entry";
+                    begin
+                        AITTestSuite.SetLoadFields("Base Version");
+                        AITTestSuite.Get(Rec."Test Suite Code");
+                        AITLogEntry.DrillDownFailedAITLogEntries(Rec."Test Suite Code", Rec."Line No.", AITTestSuite."Base Version");
+                    end;
+                }
+                field("No. of Operations - Base"; Rec."No. of Operations - Base")
+                {
+                    ApplicationArea = All;
+                    ToolTip = 'Specifies the number of operations in the base Version.';
+                    Visible = false;
+                    Enabled = false;
+                }
+                field(DurationBase; Rec."Total Duration - Base (ms)")
+                {
+                    ToolTip = 'Specifies Total Duration of the AIT for this role for the base version.';
+                    Caption = 'Total Duration Base (ms)';
+                    ApplicationArea = All;
+                    Visible = false;
+                }
+                field(AvgDurationBase; GetAvg(Rec."No. of Tests - Base", Rec."Total Duration - Base (ms)"))
+                {
+                    ToolTip = 'Specifies average duration of the AIT for this role for the base version.';
+                    Caption = 'Average Duration Base (ms)';
+                    ApplicationArea = All;
+                    Visible = false;
+                }
+                field(AvgDurationDeltaPct; GetDiffPct(GetAvg(Rec."No. of Tests - Base", Rec."Total Duration - Base (ms)"), GetAvg(Rec."No. of Tests", Rec."Total Duration (ms)")))
+                {
+                    ToolTip = 'Specifies difference in duration of the AIT for this role compared to the base version.';
+                    Caption = 'Change in Duration (%)';
+                    ApplicationArea = All;
+                    Visible = false;
+                }
+            }
+        }
+    }
+    actions
+    {
+        area(Processing)
+        {
+            action("Run Test")
+            {
+                ApplicationArea = All;
+                Caption = 'Run Test';
+                Image = Start;
+                Tooltip = 'Starts running the AIT Line.';
+
+                trigger OnAction()
+                begin
+                    if Rec."Codeunit ID" = 0 then
```

Comment:
Would it make sense to move this check into the AITTestSuiteMgt.RunAITestLine(Rec, true); procedure?

Reply:
We can keep it here, it is UI related. Codeunti should probably throw an error